unit_tests_quick_start_guide.rst

```diff
@@ -0,0 +1,173 @@
+
+Example of unit tests
+=====================
+
+Let's assume that there is the following function in ``my_module``:
+
+.. code:: python
+
+  def convert_to_supported(val):
+      """Convert unsupported types to appropriate."""
+      if isinstance(val, decimal.Decimal):
+          return float(val)
+
+      if isinstance(val, datetime.timedelta):
+          return str(val)
+
+      if val == 42:
+          raise ValueError("This number is just too cool for us ;)")
+
+      return val
+
+To test this function, we should, at least, check:
+
+* If the function gets a ``Decimal`` argument, it returns a corresponding ``float`` value.
+* If the function gets a ``timedelta`` argument, it returns a corresponding ``str`` value.
+* If the function gets ``42`` as an argument, it raises a ``ValueError``.
+* If the function gets an argument of any other type, it does nothing and returns the same value.
+
+Let's write a simple test.
+
+Assuming that our collection is called ``community.mycollection``:
+
+1. If you already have your local environment `prepared <https://github.com/ansible/community-docs/blob/main/create_pr_quick_start_guide.rst#prepare-your-environment>`_, go to the collection's root directory:
+
+.. code:: bash
+
+   cd ~/ansible_collection/community/mycollection
+
+2. Create a test file for ``my_module`` (if the following path does not exist, create it):
+
+.. code:: bash
+
+    touch tests/units/plugins/modules/test_my_module.py
+
+3. Add the following code to the file:
+
+.. code:: python
+
+  # -*- coding: utf-8 -*-
+
+  from __future__ import (absolute_import, division, print_function)
+  __metaclass__ = type
+
+  from datetime import timedelta
+  from decimal import Decimal
+
+  import pytest
+
+  from ansible_collections.community.mycollection.plugins.modules.my_module import (
+      convert_to_supported,
+  )
+
+  # We use the @pytest.mark.parametrize decorator to parametrize the function
+  # https://docs.pytest.org/en/latest/how-to/parametrize.html
+  # Simply put, the first element of each tuple will be passed to
+  # the test_convert_to_supported function as the test_input argument
+  # and the second element of each tuple will be passed as
+  # the expected argument.
+  # In the function's body, we use the assert statement to check
+  # if the convert_to_supported function given the test_input,
+  # returns what we expect.
+  @pytest.mark.parametrize('test_input, expected', [
+      (timedelta(0, 43200), '12:00:00'),
+      (Decimal('1.01'), 1.01),
+      ('string', 'string'),
+      (None, None),
+      (1, 1),
+  ])
+  def test_convert_to_supported(test_input, expected):
+      assert convert_to_supported(test_input) == expected
+
+  def test_convert_to_supported_exception():
+      with pytest.raises(ValueError, match=r"*too cool*"):
```

Comment:
The `match` parameter should contain a valid regular expression or compiled pattern, but this example contains none of those things. Are we sure this example works? And is there a reason why we need those `*` around the string we are trying to match?

Reply:
Ah, forgot to add dots (this format is used on pytest website, i thought without the starts it would try to check if the err msg is completely equal to the quoted expression).. I tested, it works without stars and dots as well. Good catch, fixed, thanks!